lib/inc/drogon/utils/Utilities.h

```diff
@@ -58,10 +58,10 @@ struct CanConvertFromStringStream
 namespace utils
 {
 /// Determine if the string is an integer
-DROGON_EXPORT bool isInteger(const std::string &str);
+DROGON_EXPORT bool isInteger(string_view str);
```

Comment:
Please add two new interface instead of replacing the old ones.

Reply:
String_views encompass `const char*`, `const char[N]`, `std::string` and `std::string_view`, is there a need for new interfaces? It doesn't break old code using these methods.
There is a string_view class built into drogon, it won't break compatibility with C++ versions prior to C++17 as well.

Reply:
What I mean is, considering the scenario of using a std::string object as a parameter when calling the function, if we provide multiple versions, we can avoid explicitly or implicitly constructing string_view objects.

Reply:
Assuming heap allocation

`const string&`:
- itself being a pointer, 8 bytes
- data char pointer, 8 bytes
- length as size_t, 8 bytes
- one layer of indirection to access any of its members
- two layers of indirection to access its data
- if an `std::string` was passed to the function, we will be using a total of 32 bytes (the only positive)

`string_view`:
- data char pointer, 8 bytes
- length as size_t, 8 bytes
- no indirection in accessing any of its members
- one layer of indirection to access its data
- if an `std::string` was passed to construct a `string_view`, we will be using a total of 40 bytes

`string_view`'s construction is trivial. It is C++, a matter of jamming the 16 bytes into a register or somewhere in memory.

I'm an advocate of string_views over `const string&`, I may be biased.
Before I start making changes, what is your final say, string or string_view?

Reply:
If we already have a std::string object (some string created dynamically), I don't need to create a string_view if we have a string reference version. so I prefer to previde two versions of this function, one accepts const string reference and another accepts string_view value.

Reply:
@an-tao @Mis1eader-dev 

The construction of `string_view` is quite negligible since compilers can easily optimize that out due to its simplicity. The following link shows that the function using `const string&` has extra instruction calls than the function using `string_view`.

https://godbolt.org/z/WKG1cezs6

Reply:
What happens if string_view is boost::string_view when using c++14?

Reply:
Shall I make a pull request removing the `const string&` versions, or leave it?

Reply:
@Mis1eader-dev (cc @an-tao)
Yes, could you please remove the `const string&` implementations because of no difference on C++14 but improvements in later versions?

Reply:
Done #1693